pkg/apis/provisioning/v1alpha5/suite_test.go

```diff
@@ -211,7 +211,22 @@ var _ = Describe("Validation", func() {
 			B := NewRequirements(v1.NodeSelectorRequirement{Key: v1.LabelTopologyZone, Operator: v1.NodeSelectorOpIn, Values: []string{"foo"}})
 			Expect(A.Compatible(B)).To(Succeed())
 		})
-		It("A should fail to be compatible to B, <In, In> operaton, no overlap", func() {
+		It("A should be compatible to B, <In, In> operator, A is empty list", func() {
+			// We're asking here ifs zone [] compatible with zone ["foo"].  K8s itself won't allow this to be configured on a pod
+			// as it requires values if you specify operator In or NotIn and won't allow an empty list. Logically we consider an empty set
+			// to be a subset of any other set.
+			A := NewRequirements(v1.NodeSelectorRequirement{Key: v1.LabelTopologyZone, Operator: v1.NodeSelectorOpIn, Values: []string{}})
```

Comment:
I'm not sure I agree with this. There's a difference with a rule not being `nil` and a rule being defined as `[]`.

Reply:
Users can't create the empty rule since k8s prevents it.  The code a few lines above was similar in that it prevents an empty rule from preventing a match. 

```go
if values := requirements.Get(key); values.Len() != 0 && !values.IsComplement() && !r.hasRequirement(withKey(key)) {
			errs = multierr.Append(errs, fmt.Errorf("require values for key %s but is not defined", key))
		}
```

 Since users can't create these types of rules, another option is to go modify/fix our code so it stops putting empty rules in.

Reply:
> another option is to go modify/fix our code so it stops putting empty rules in.

Sounds good. This sounds like an invariant is being violated.